packages/cli/src/services/project.service.ts

```diff
@@ -59,24 +65,43 @@ export class ProjectService {
 			);
 		}
 
-		// 1. disable all workflows
+		let targetProject: Project | null = null;
+		if (migrateToProject) {
+			targetProject = await this.getProjectWithScope(user, migrateToProject, [
+				'credential:create',
+				'workflow:create',
+			]);
+
+			if (!targetProject) {
+				throw new NotFoundError(
+					`Could not find project to migrate to. ID: ${targetProject}. You may lack permissions to create workflow and credentials in the target project.`,
+				);
+			}
+		}
 
+		// 1. disable all workflows (unless they are migrated)
 		const ownedSharedWorkflows = await this.sharedWorkflowRepository.find({
 			where: { projectId: project.id, role: 'workflow:owner' },
 			relations: { workflow: true },
 		});
 
-		// NOTE: This is supposed to happen outside the transaction.
-		// We'd otherwise need to pass the em through to a lot of functions even ending up in core.
-		// See: https://github.com/n8n-io/n8n/pull/8904#discussion_r1530150510
-		for (const sharedWorkflow of ownedSharedWorkflows) {
-			await activeWorkflowRunner.remove(sharedWorkflow.workflow.id);
+		if (!targetProject) {
+			// NOTE: This is supposed to happen outside the transaction.
+			// We'd otherwise need to pass the em through to a lot of functions even ending up in core.
+			// See: https://github.com/n8n-io/n8n/pull/8904#discussion_r1530150510
+			for (const sharedWorkflow of ownedSharedWorkflows) {
+				await activeWorkflowRunner.remove(sharedWorkflow.workflow.id);
+			}
 		}
 
 		await this.projectRelationRepository.manager.transaction(async (em) => {
```

Comment:
`tx` or `trx` is a common name to use for a variable that represents a transaction 

```suggestion
		await this.projectRelationRepository.manager.transaction(async (trx) => {
```

Reply:
I don't have strong opinions either if the value should be `trx`, `tx` or `transactionManager` as long as it's used in a unified fashion. IMO the name should reflect that it's an EntityManager that is tied to an explicit transaction (That's what `manager.transaction` actually does. It creates an EntityManager with a reference to a QueryRunner (i.e. DB connection) and begins a transaction in that connection).

We should unify this in the whole codebase. Doing it after the feature branch is merged makes sense 👍 Let's use the  correct name already in new code so we don't create more debt.